medallion/test/config/test_config_loading.py

```diff
@@ -0,0 +1,353 @@
+    """
+    Confirm that nothing bad happens if the default config file is missing.
+    """
+    with mock.patch(
+        "medallion.config.DEFAULT_CONFFILE", str(tmp_path / "medallion.conf"),
+    ) as mock_conf_file_default:
+        with subtests.test(msg="conf_dir omitted"):
+            config = m_cfg.load_config(conf_file=mock_conf_file_default)
+            assert config == {}
+        with subtests.test(msg="conf_dir is None"):
+            config = m_cfg.load_config(
+                conf_file=mock_conf_file_default, conf_dir=None,
+            )
+            assert config == {}
+
+
+def test_load_config_non_default_file_missing(subtests, tmp_path):
+    """
+    Confirm an exception is raised if a non-default config file is missing.
+    """
+    conf_file = tmp_path / "missing.conf"
+    exc_patt = re.escape(repr(str(conf_file)))
+    with pytest.raises(FileNotFoundError, match=exc_patt):
+        m_cfg.load_config(conf_file=conf_file)
+
+
+@mock.patch.dict("os.environ", clear=True)
+def test_load_config_default_dir_missing(subtests, tmp_path):
+    """
+    Confirm that nothing bad happens if the default config dir is missing.
+    """
+    with mock.patch(
+        "medallion.config.DEFAULT_CONFDIR", str(tmp_path / "medallion.d"),
+    ) as mock_conf_dir_default:
+        with subtests.test(msg="conf_file omitted"):
+            config = m_cfg.load_config(conf_dir=mock_conf_dir_default,)
+            assert config == {}
+        with subtests.test(msg="conf_file is None"):
+            config = m_cfg.load_config(
+                conf_dir=mock_conf_dir_default, conf_file=None,
+            )
+            assert config == {}
+
+
+def test_load_config_non_default_dir_missing(tmp_path):
+    """
+    Confirm an exception is raised if a non-default config dir is missing.
+    """
+    conf_dir = tmp_path / "missing.d"
+    exc_patt = re.escape(repr(str(conf_dir)))
+    with pytest.raises(FileNotFoundError, match=exc_patt):
+        m_cfg.load_config(conf_dir=conf_dir)
+
+
+@mock.patch.dict("os.environ", clear=True)
+def test_load_config_logging(subtests, tmp_path, caplog):
+    """
+    Ensure that config loading emits log messages about paths and data.
+    """
+    confdata = {"foo": "bar"}
+    conf_file = tmp_path / "medallion.conf"
+    json.dump(confdata, conf_file.open("w"))
+    caplog.set_level(logging.DEBUG, logger="medallion.config")
+    with subtests.test(msg="Used as config file"):
+        assert m_cfg.load_config(conf_file=conf_file) == confdata
+        assert f"load configuration from {str(conf_file)!r}" in caplog.text
+        for record in caplog.records:
+            if '"foo": "bar"' in record.message:
+                break
+        else:
+            raise AssertionError("Config data not seen in log output")
+    caplog.clear()
+    with subtests.test(msg="Used from config dir"):
+        assert m_cfg.load_config(conf_dir=tmp_path) == confdata
+        assert f"and {str(tmp_path)!r}" in caplog.text
```

Comment:
Similar to above, the left side is printing with more slashes than the right on Windows (aren't paths between different OSes wonderful?)

Reply:
As above, please resolve if it's fixed!